mcos/covariance_transformer.py

```diff
@@ -0,0 +1,211 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from sklearn.neighbors import KernelDensity
+from scipy.optimize import minimize
+import pandas as pd
+
+
+class AbstractCovarianceTransformer(ABC):
+    """
+    Abstract class for transforming a covariance matrix
+    """
+
+    @abstractmethod
+    def transform(self, cov: np.array, n_observations: int) -> np.array:
+        """
+        Transforms a covariance matrix
+        :param cov: covariance matrix
+        :param n_observations: number of observations used to create the covariance matrix
+        :return: transformed covariance matrix
+        """
+
+        pass
+
+
+class CovarianceMatrixDeNoiser(AbstractCovarianceTransformer):
```

Comment:
Could you rename this as DeNoiserCovarianceTransformer, so it's easily identifiable as a CovarianceTransformer?

Reply:
for sure!